sky_templates/ray/start_cluster.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+# Starts a Ray cluster on a SkyPilot cluster.
+#
+# This script starts a Ray cluster using default Ray ports (6379, 8265),
+# which are different from SkyPilot's system Ray ports (6380, 8266).
+# This allows users to run their own Ray applications independently of
+# SkyPilot's internal Ray cluster.
+#
+# Environment Variables:
+#   RAY_HEAD_PORT=6379                     - Ray head node port
+#   RAY_DASHBOARD_PORT=8265                - Ray dashboard port
+#   RAY_DASHBOARD_HOST=127.0.0.1           - Dashboard host (set to 0.0.0.0 to expose externally)
+#   RAY_DASHBOARD_AGENT_LISTEN_PORT=       - (Optional) Dashboard agent listen port
+#   RAY_NODE_IP_ADDRESS=                   - (Optional) Node IP address
+#   RAY_CMD_PREFIX=                        - (Optional) Command prefix (e.g., "uv run")
```

Comment:
These are the minimal set of params needed to cover all our examples that uses Ray.